protocols/virtualdefund/virtualdefund.go

```diff
@@ -115,67 +124,81 @@ func NewObjective(request ObjectiveRequest,
 		return Objective{}, fmt.Errorf("client address not found in an expected participant index")
 
 	}
+
+	// If we can't find any vouchers we use an unsigned voucher of 0
+	// TODO: We have a "zero" voucher and now a voucher with amount 0
+	latestVoucher := *payments.NewVoucher(request.ChannelId, big.NewInt(0))
+	if myAddress == alice {
+		var err error
+		latestVoucher, err = getVoucher(request.ChannelId)
+		if err != nil {
+			return Objective{}, fmt.Errorf("could not get the latest voucher for channel %s: %w", request.ChannelId, err)
+		}
+
+	}
+
+	vouchers := [3]*payments.Voucher{}
+	vouchers[V.MyIndex] = &latestVoucher
 	return Objective{
 		Status:         status,
-		InitialOutcome: initialOutcome,
-		PaidToBob:      request.PaidToBob,
-		VFixed:         V.FixedPart,
-		Signatures:     [3]state.Signature{},
-		MyRole:         V.MyIndex,
-		ToMyLeft:       toMyLeft,
-		ToMyRight:      toMyRight,
+		InitialOutcome: initialOutcome, VFixed: V.FixedPart,
+		Signatures: [3]state.Signature{},
+		MyRole:     V.MyIndex,
+		ToMyLeft:   toMyLeft,
+		ToMyRight:  toMyRight,
+		Vouchers:   vouchers,
 	}, nil
 
 }
 
-// ConstructObjectiveFromState takes in a message and constructs an objective from it.
+// ConstructObjectiveFromVoucher takes in a message and constructs an objective from it.
 // It accepts the message, myAddress, and a function to to retrieve ledgers from a store.
-func ConstructObjectiveFromState(
-	initialState state.State,
+func ConstructObjectiveFromVoucher(
+	fixedPart state.FixedPart,
+	initialVoucher payments.Voucher,
```

Comment:
having an `initialVoucher` and a `getVoucher` is slightly confusing. We should rename this to make it clear that `initialVoucher` is supposed to be a voucher we received in a message.

Reply:
Addressed in [d5f02ae](https://github.com/statechannels/go-nitro/pull/826/commits/d5f02ae41aec53db1b424869e26b90e74621412c)